genesyscloud/routing_email_route/genesyscloud_routing_email_route_proxy.go

```diff
@@ -92,73 +92,65 @@ func (p *routingEmailRouteProxy) deleteRoutingEmailRoute(ctx context.Context, do
-	var allInboundRoutes = make(map[string][]platformclientv2.Inboundroute)
+func getAllRoutingEmailRouteFn(ctx context.Context, p *routingEmailRouteProxy, domainId, pattern string) (*map[string][]platformclientv2.Inboundroute, *platformclientv2.APIResponse, error) {
 	const pageSize = 100
 	var apiResponse *platformclientv2.APIResponse
 
-	domains, resp, err := p.routingApi.GetRoutingEmailDomains(pageSize, 1, false, domainId)
-	if err != nil {
-		return nil, resp, fmt.Errorf("Failed to get routing email domains: %s", err)
+	var allDomains = make([]platformclientv2.Inbounddomain, 0)
+	for pageNum := 1; ; pageNum++ {
+		domains, resp, err := p.routingApi.GetRoutingEmailDomains(pageSize, pageNum, false, domainId)
+		apiResponse = resp
+		if err != nil {
+			return nil, resp, fmt.Errorf("failed to get routing email domains: %s", err.Error())
+		}
+		if domains.Entities == nil || len(*domains.Entities) == 0 {
+			break
+		}
+		allDomains = append(allDomains, *domains.Entities...)
 	}
-	if domains.Entities == nil || len(*domains.Entities) == 0 {
-		return &allInboundRoutes, resp, nil
+
+	if len(allDomains) == 0 {
+		return nil, apiResponse, nil
 	}
 
-	// DomainID not given so we must acquire every route for every domain
-	routes, _, err := getAllRoutingEmailRouteByDomainIdFn(ctx, p, domains, name)
+	// Get all routes for each domain
+	routes, resp, err := getAllRoutingEmailRouteByDomainIdFn(ctx, p, allDomains, pattern)
 	if err != nil {
-		return nil, resp, fmt.Errorf("Failed to get routing email domains: %s", err)
+		return nil, resp, fmt.Errorf("failed to get routing email domains: %s", err.Error())
 	}
 
-	allInboundRoutes = *routes
-	// If domainID is given, we only return the routes for that specific domain
-	if domainId != "" {
-		log.Printf("Returning routes for domain ID: %s", domainId)
-		return &allInboundRoutes, apiResponse, nil
+	if routes == nil {
+		log.Printf("No routing email routes found. domainId: '%s', pattern: '%s'", domainId, pattern)
+		return nil, resp, nil
 	}
 
-	// DomainID not given so we must acquire every route for every domain
-	for pageNum := 2; pageNum <= *domains.PageCount; pageNum++ {
-		domains, resp, err := p.routingApi.GetRoutingEmailDomains(pageSize, pageNum, false, "")
-		if err != nil {
-			return nil, resp, fmt.Errorf("Failed to get routing email domains: %s", err)
-		}
-		if domains.Entities == nil || len(*domains.Entities) == 0 {
-			return &allInboundRoutes, resp, nil
-		}
-
-		routes, _, err := getAllRoutingEmailRouteByDomainIdFn(ctx, p, domains, name)
-		if err != nil {
-			return nil, resp, fmt.Errorf("Failed to get routing email domains: %s", err)
-		}
-		mergeIntoExistingMap(allInboundRoutes, *routes)
-	}
-	log.Printf("Returning routes for domains: %v", reflect.ValueOf(allInboundRoutes).MapKeys())
-	return &allInboundRoutes, apiResponse, nil
```

Comment:
This was unnecessary because it just did the same thing accept with `""` instead of `domainId` , and only if domainId was "". But if domainId was "", then we were doing the same thing twice; once with the empty string variable and again with a hard-coded empty string.